examples/download.rs

```diff
@@ -102,4 +102,22 @@ fn main() {
             Ok(s) => println!("Success: {}", &s),
         };
     }
+
+    let debian_mirrors = MirrorContext::from_urls(&[
+        "http://ftp.au.debian.org/debian/",
+        "http://ftp.ca.debian.org/debian/",
+        "http://ftp.cz.debian.org/debian/",
+        "http://ftp.us.debian.org/debian/",
+    ]).unwrap();
+
+    let dl = downloader::Download::new_mirrored(debian_mirrors.urls_for_file("README.html").unwrap().as_slice());
```

Comment:
I'd like a separate method which takes a context directly

Reply:
I am not sure that is a good idea wrt. "keep this simple".

How about making the function always take an `Option<MirrorContext>` and documenting that is fine to leave it as `None` if you have a bunch of (non-relative) URLs.